README.md

```diff
@@ -30,35 +51,210 @@ npm install --save ipfsd-ctl
+
+const port = 9999
+const server = DaemonFactory.createServer(port)
+const df = DaemonFactory.create({ remote: true, port: port })
+
+server.start((err) => {
+  if (err) { throw err }
+
+  df.spawn((err, ipfsd) => {
+    if (err) { throw err }
+
+    ipfsd.api.id(function (err, id) {
+      if (err) { throw err }
+      
+      console.log(id)
+      ipfsd.stop(server.stop)
+    })
+  })
+})
+```
+
+## Disposable vs non Disposable nodes
+
+`ipfsd-ctl` can create two types of node controllers, `disposable` and `non-disposable`. A disposable node will be created on a temporary repo which will be optionally initialized and started (the default), as well cleaned up on process exit. A non-disposable node on the other hand, requires the user to initialize and start the node, as well as stop and cleanup after wards. Additionally, a non-disposable will allow you to pass a custom repo using the `repoPath` option, if the `repoPath` is not defined, it will use the default repo for the node type (`$HOME/.ipfs` or `$HOME/.jsipfs`). The `repoPath` parameter is ignored for disposable nodes, as there is a risk of deleting a live repo.
+
+## IPFS executables
+
+`ipfsd-ctl` no longer installs go-ipfs nor js-ipfs dependencies, instead it expects them to be provided by the parent project. In order to be able to use both go and js daemons, please make sure that your project includes these two npm packages as dependencies.
+
+- `ipfs` - the js-ipfs implementation
+- `go-ipfs-dep` - the packaged go-ipfs implementation
+
+## API
+
+### Daemon Factory Class
+
+#### `DaemonFactory` - `const df = DaemonFactory.create([options])`
+
+`DaemonFactory.create([options])` returns an object that will expose the `df.spawn` method
+
+- `options` - an optional object with the following properties
+  - `remote` bool - indicates if the factory should spawn local or remote nodes. By default, local nodes are spawned in Node.js and remote nodes are spawned in Browser environments.
+  - `port` number - the port number to use for the remote factory. It should match the port on which `DaemonFactory.server` was started. Defaults to 9999.
+  - `type` - the daemon type to create with this factory. See the section bellow for the supported types
+  - `exec` - path to the desired IPFS executable to spawn, otherwise `ipfsd-ctl` will try to locate the correct one based on the `type`. In the case of `proc` type, exec is required and expects an IPFS coderef.
+  
+`ipfsd-ctl` allows spawning different IPFS implementations, such as:
+
+- **`go`** - calling `DaemonFactory.create({type: 'go'})` will spawn a `go-ipfs` daemon.
+- **`js`** - calling `DaemonFactory.create({type: 'js'})` will spawn a `js-ipfs` daemon.
+- **`proc`** - calling `DaemonFactory.create({type: 'proc', exec: require('ipfs') })` will spawn an `in process js-ipfs node` using the provided code reference that implements the core IPFS API. Note that, `exec` option to `df.spawn()` is required if `type: 'proc'` is used.
```

Comment:
That would force users to bring `js-ipfs` at all times and we learned that makes it hard for users that want to bundle go-ipfs in. Otherwise, it would be good.

Reply:
Agreed.

Reply:
Let's call this `embedded` instead? Seems more consistent with e.g. ipfs-companion. (`proc` is also a pretty generic name.)